### ISSUE TITLE
Add a command to open a new window as tab (Sierra tabs)

### ISSUE THREAD

**Issue by miltonbecker** [Microsoft/vscode#25919]
- VSCode Version: 1.12
- OS Version: macOS Sierra

Steps to Reproduce:

1. Enable window.nativeTabs in settings
2. Click on View -> Show Tab Bar
3. Try to open a new tab now - there is no File -> New Tab, pressing cmd+shift+t or cmd+t doesn't open a tab either, double clicking the tab also doesn't do anything.

The only way to use the new native tab on macOS is to open a new vscode window then drag one tab into another window.

**Comment by kieferrm** [comment]
@miltonbecker the behavior `New window` depends on your OS setting.

**Comment by miltonbecker** [comment]
@kieferrm what setting specifically?

**Comment by gdad-s-river** [comment]
⌘ + ⇧ + n does it for me

**Comment by miltonbecker** [comment]
@gdad-s-river that opens a new window here. I'm wondering what OS setting I have to change :/

**Comment by Richard-Scap** [comment]
I'm having the same issue. Code version 1.12, OS version 10.12.3 

Also, any reason why 'Show Tab Bar' is not available in the Command Palette?

**Comment by sgnh** [comment]
You can change this under "System Preferenes" -> "Dock".
<img width="668" alt="dock" src="https://cloud.githubusercontent.com/assets/4149159/25727452/9851d7a0-3129-11e7-856b-43adf3b987dd.png">

**Comment by miltonbecker** [comment]
@sgnh that works, thanks!

But now there is another problem: I can't open a new window 😆 
So it's either window OR tab.

I believe there should be a "File -> New Tab" menu option and a "+" button so we can add more tabs easily.

Anyway, should I close this issue or leave it as a feature request?

**Comment by jaybeecave** [comment]
This was working as expected in 1.12. All new windows would open in new tabs.. 1.12.1 its broken again. Now when I open 2 projects from command line they are back to opening in seperate windows...

**Comment by HADB** [comment]
There's no "Show Tab Bar" in 1.12.1 😭

**Comment by HADB** [comment]
I found the problem.
If set "window.newWindowDimensions" to "fullscreen", the new window will not show as tab bar.

**Comment by balupton** [comment]
this is totally not what I was expecting by this... I was expecting that tabs would now use the native tab bar - not windows

**Comment by gianmarcotoso** [comment]
I would like to add that calling `code` on a folder/file from the command line opens a new window regardless of the `window.nativeTabs` setting.

**Comment by bpasero** [comment]
Our intention was never to replace the editor tabs with native tabs. Native tabs are to group windows. It is true that we might provide limited functionality, hence we should collect more issues in this item and then propose changes to Electron 👍

**Comment by yingbo** [comment]
It's very confusing to have both "native tabs" and "editor tabs". I think giving users an option to replace editor tabs by the native one is great. Otherwise, the native tabs is, at least to me, not useful at all.

**Comment by gianmarcotoso** [comment]
@yingbo For me native tabs play a different roles than editor tabs: I use native tabs to have multiple projects opened, while editor tabs (actually, I keep open files on the sidebar) for the single project's files. This way I don't need to have multiple VSCode windows open!

**Comment by davistan** [comment]
In File menu, should have New Tab and New Window (like in Chrome). There will be New File menu which is just adding a tab in editor.

For Window menu, it should have "Merge All Windows". If you're using SourceTree you know what I meant.

And to make it more usable, adding a + icon beside the native tabs will be natural for UX. Refer to Safari browser.

This is my 2 cents. Hope this helps.

**Comment by kethinov** [comment]
For anyone struggling to get View > Show Tab Bar to appear, pasting `"window.nativeTabs": true` settings.json and manually quitting and reopening the app doesn't appear to work. You have to fish for it under "Search Settings" on the left panel via Preferences > Settings, click the edit icon, and set it to true that way. That will have the effect of pasting `"window.nativeTabs": true` into settings.json, but it will also trigger code to *prompt* you to restart the editor, after which if you answer yes, it will actually display the new View > Show Tab Bar option.

**Comment by LeoNatan** [comment]
The current implementation goes against Cocoa design guidelines for the tabs, thus extremely confusing and one has to wonder what was the point of implementing in the first place. Tabs are meant for document apps to be able to collect multiple windows (each displaying a document). Sounds a lot like “editor tabs”. There should be an option to replace editor tabs completely, if this is what users prefer. I reckon most Mac users would prefer using native tabs rather than “editor tabs”.

**Comment by qbilius** [comment]
Native tabs work just fine for me in 1.13.1 but after restarting VS Code all these tabs open in new separate windows. Since there is no "Merge All Windows" option, the only way to put them back into a single window is drag and drop, which is really inconvenient.

**Comment by djalmaaraujo** [comment]
How can we open a new tab? I know that the `new-window-for-tab` even is available now, but can we force the event to be called? Right now we can't create a shortcut like cmd+t, correct? Is this one of the things this PR will solve?

**Comment by beeing** [comment]
I thought I saw merge all windows to tabs, but now in 1.14.0 it's gone.

**Comment by okonon** [comment]
I can't get native tabs to work either. Version 1.14.2 (1.14.2)

**Comment by bishless** [comment]
If I pair these two settings, I can get native tabs to work in 1.15.0.

```
"window.nativeTabs": true,
"window.openFoldersInNewWindow": "on",
```

**Comment by DavidBabel** [comment]
It worked for me for a long time, but after rewrite my config (for cleanup) it broke and it's now impossible to get native tabs back again.

Sierra 10.12.6
vscode 1.15.1

If someone found a workaround to get this work i'm very interested, since it's a very usefull functionnality on MacOS where CMD+tab does not list window of the same app.

**Comment by bpasero** [comment]
Latest VS Code insider releases (https://code.visualstudio.com/insiders) come with some additional features for native tab support:
* there is a "+" in the tab well to create a new tab (= empty window)
* there are new commands to manage tabs under the window menu (Show Next Tab, Show Previous Tab, Merge All Windows, Move Tab to New Window) 
* all commands are also accessible from command palette and can have keybindings (including the command Toggle Windows Tabs Bar to show or hide the tab bar)

See https://twitter.com/BenjaminPasero/status/906162846186209281

**Comment by miltonbecker** [comment]
The new stable release (1.17.0) has completely broken the native tabs support on macOS High Sierra (10.13).

- The tab bar is now blank, completely blank. I can't see any tab, nor text, nor the + sign.
- When I click on the supposed-to-be tab bar, a new window opens instead.
- When I click on "Merge All Windows", I can't see any tabs on the tab bar. However, I can see the tabs when I click on "Show All Tabs".

**Comment by bpasero** [comment]
High Sierra is covered in https://github.com/Microsoft/vscode/issues/35361

**Comment by qiuyuntao** [comment]
In my vscode, tabbar is still blank.

**Comment by DavidBabel** [comment]
Try to update to 1.19.2

**Comment by miltonbecker** [comment]
@qiuyuntao @DavidBabel this issue is caused by an electron issue. You can see the electron issue at https://github.com/electron/electron/issues/10657.

We can only hope that issue gets fixed soon :(

**Comment by jatinwaichal** [comment]
Hi, I am not sure if this issue is fixed or not. 
I am using Mac OS High Sierra 10.13.4 & Visual Studio 1.23.1.
I tried setting the nativtab to true, and other setting mentioned above. But it is still not working.
I know it was working before.

Any help or guidance I highly appreciated.

**Comment by miltonbecker** [comment]
@jatinwaichal the source issue was apparently fixed on electron 2.0. 
So now we're waiting for vscode to upgrade its electron from 1.7 to 2.0 😄
You can follow this process here: https://github.com/Microsoft/vscode/issues/45542

**Comment by jatinwaichal** [comment]
thanks, @miltonbecker for the update. I will keep an eye on the other ticket

**Comment by miltonbecker** [comment]
bpasero found a fix! It's still not official but I can confirm it works! (and in the end this issue might not have anything to do with electron hah)

you can check the comments here: https://github.com/Microsoft/vscode/issues/35361#issuecomment-395365519

and the quick-fix-command as said by johnelm:
`sed -i -- 's/com.microsoft.VSCode/com.microsoft2.VSCode/g' /Applications/Visual\ Studio\ Code.app/Contents/Info.plist`

**Comment by bpasero** [comment]
With Electron 2.0.x we can have a new "New Tab" entry under the "Window" menu to open a new window as window tab:

![image](https://user-images.githubusercontent.com/900690/43787915-707eb0a6-9a6c-11e8-9357-f57d5b82647d.png)

There is also a new command to assign a keybinding to: `workbench.action.newWindowTab`

**Verifier:**
* you need to enable `"window.nativeTabs": true` and restart
* you can use the command from the Window menu
* you can use the command via keybinding or command palette

**Comment by spike1292** [comment]
@bpasero Awesome! The new tab is working like a charm in the latest insiders build.

Just installed vscode insiders on a new Mac 

<img width="1024" alt="screen shot 2018-08-18 at 17 57 30" src="https://user-images.githubusercontent.com/288597/44301025-d7cada80-a310-11e8-8841-bc214837023b.png">

and it's not showing the tabs any more. I didn't see this problem when I updated from 1.25 to 1.26. 

So I think electron 2.0.x broke setting the `NSUseImprovedLayoutPass` default, see https://github.com/Microsoft/vscode/blob/b6411f27b0e41588fa897d51c34eafb636f6cbae/src/vs/code/electron-main/app.ts#L293

After I set the defaults manually:
```sh
defaults write com.microsoft.VSCodeInsiders NSUseImprovedLayoutPass -bool true
```
The tabs show up!
<img width="1024" alt="screen shot 2018-08-18 at 18 08 29" src="https://user-images.githubusercontent.com/288597/44301103-d221c480-a311-11e8-8a3b-1e5a4538ec6d.png">
